src/rest/AnalyticsEndpoint.ts

```diff
@@ -47,6 +48,15 @@ export class AnalyticsEndpoint {
     this.organization = options.organization;
   }
 
+  public static getURLFromSearchEndpoint(endpoint: SearchEndpoint): string {
+    if (!endpoint || !endpoint.options || !endpoint.options.restUri) {
+      return this.DEFAULT_ANALYTICS_URI;
+    }
+
+    const [basePlatform] = endpoint.options.restUri.split('/rest');
```

Comment:
Is it possible for a url to either not contain `/rest` or multiple `/rest` (I remember Salesforce using a proxy)? If the goal is to extract the domain name, we could do it without assuming anything about the path. e.g.

```
const a = document.createElement('a');
a.href = endpoint.options.restUri;

return `${link.protocol}//${link.host}/rest/ua`;
```

In either approach though, should we handle the case where the uri is either missing a `/rest` or is invalid?